x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/use_bulk_actions.tsx

```diff
@@ -72,6 +72,8 @@ export const useBulkActions = ({
   const getIsMounted = useIsMounted();
   const filterQuery = convertRulesFilterToKQL(filterOptions);
   const { startTransaction } = useStartTransaction();
+  const [bulkAction, setBulkAction] = useState<BulkAction | undefined>(undefined);
+  const [bulkActionEdit, setBulkActionEdit] = useState<BulkActionEditType | undefined>(undefined);
```

Comment:
What is the purpose of these two states? Imperative actions should not use state to communicate with their parent component. That probably indicates that data fetching callbacks were designed incorrectly.